core/src/main/java/org/axonframework/config/EventProcessingModule.java

```diff
@@ -0,0 +1,682 @@
+        if (configuration != null) {
+            eventProcessor(processorName).ifPresent(eventProcessor -> eventProcessor
+                    .registerHandlerInterceptor(interceptorBuilder.apply(configuration)));
+        }
+        this.handlerInterceptorsBuilders.computeIfAbsent(processorName, k -> new ArrayList<>())
+                                        .add(interceptorBuilder);
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfigurer registerDefaultHandlerInterceptor(
+            BiFunction<Configuration, String, MessageHandlerInterceptor<? super EventMessage<?>>> interceptorBuilder) {
+        this.defaultHandlerInterceptors.add(interceptorBuilder);
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfigurer registerSequencingPolicy(String processingGroup,
+                                                              Function<Configuration, SequencingPolicy<? super EventMessage<?>>> policyBuilder) {
+        this.sequencingPolicies.put(processingGroup, new Component<>(() -> configuration,
+                                                                     "sequencingPolicy",
+                                                                     policyBuilder));
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfigurer registerDefaultSequencingPolicy(
+            Function<Configuration, SequencingPolicy<? super EventMessage<?>>> policyBuilder) {
+        this.defaultSequencingPolicy.update(policyBuilder);
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfigurer registerMessageMonitorFactory(String name,
+                                                                   MessageMonitorFactory messageMonitorFactory) {
+        this.messageMonitorFactories.put(name, messageMonitorFactory);
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfigurer registerRollbackConfiguration(String name,
+                                                                   Function<Configuration, RollbackConfiguration> rollbackConfigurationBuilder) {
+        this.rollbackConfigurations.put(name, new Component<>(() -> configuration,
+                                                              "rollbackConfiguration",
+                                                              rollbackConfigurationBuilder));
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfigurer registerTransactionManager(String name,
+                                                                Function<Configuration, TransactionManager> transactionManagerBuilder) {
+        this.transactionManagers.put(name, new Component<>(() -> configuration,
+                                                           "transactionManager",
+                                                           transactionManagerBuilder));
+        return this;
+    }
+
+    @Override
+    public EventProcessingConfiguration configure() {
```

Comment:
For a configurer module to be built, it would need to main configuration. What's the purpose of this method?

Reply:
This method returns an instance of `EventProcessingConfiguration` - in our case the same class `EventProcessingModule` implements `EventProcessingConfigurer` and `EventProcessingConfiguration`. But, in some cases, it would make sense to have `Configuration` and `Configurer` implemented by separate classes.

Reply:
I agree with that, but why would one want to get from EventProcessingModule (in the Configurer phase) to Configuration? It doesn't make much sense, as none of the Configuration components is accessible anyway. I think it makes more sense to get the EventProcessingConfiguration from the main Configuration, where you can ensure the configuration is completed.

Reply:
Think of it as a `builder` method. You'd build `EventProcessingConfiguration` and register it with main `Configurer` which would initialize it during the initialization phase.